packages/dds/tree/src/core/tree/anchorSet.ts

```diff
@@ -769,10 +778,12 @@ export class AnchorSet implements Listenable<AnchorSetRootEvents>, AnchorLocator
 			notifyChildrenChanged(): void {
 				this.maybeWithNode(
 					(p) => {
+						assert(this.parentField !== undefined, "Must be in a field");
```

Comment:
Nit: a bit more detail in the error message would probably be useful for future debugging.